demos/text_detection_demo/cpp_gapi/main.cpp

```diff
@@ -0,0 +1,349 @@
+
+#include "gflags/gflags.h"
+#include "monitors/presenter.h"
+#include "utils/common.hpp"
+#include "utils_gapi/stream_source.hpp"
+#include "utils/performance_metrics.hpp"
+#include "utils/slog.hpp"
+
+#include "shared_functions.hpp"
+#include "nets_configuration.hpp"
+#include "text_recognition.hpp"
+#include "custom_kernels.hpp"
+#include "custom_nets.hpp"
+
+#include "text_detection_demo_gapi.hpp"
+
+static
+void setLabel(cv::Mat& im, const std::string& label, const cv::Point& p) {
+    int fontface = cv::FONT_HERSHEY_SIMPLEX;
+    double scale = 0.7;
+    int thickness = 1;
+    int baseline = 0;
+
+    cv::Size textSize = cv::getTextSize(label, fontface, scale, thickness, &baseline);
+    auto textPos = p;
+    textPos.x = std::max(0, p.x);
+    textPos.y = std::max(textSize.height, p.y);
+
+    cv::rectangle(im, textPos + cv::Point(0, baseline),
+                  textPos + cv::Point(textSize.width, -textSize.height),
+                  CV_RGB(50, 205, 50), cv::FILLED);
+    cv::putText(im, label, textPos, fontface, scale, CV_RGB(255, 255, 255), thickness, 8);
+}
+
+bool ParseAndCheckCommandLine(int argc, char* argv[]) {
+    gflags::ParseCommandLineNonHelpFlags(&argc, &argv, true);
+    if (FLAGS_h) {
+        showUsage();
+        showAvailableDevices();
+        return false;
+    }
+    if (FLAGS_i.empty()) {
+        throw std::logic_error("Parameter -i is not set");
+    }
+    if (FLAGS_m_td.empty() && FLAGS_m_tr.empty()) {
+        throw std::logic_error("Neither parameter -m_td nor -m_tr is not set");
+    }
+    if (!FLAGS_m_tr.empty() && FLAGS_dt.empty()) {
+        throw std::logic_error("Parameter -dt is not set");
+    }
+    return true;
+}
+
+int clip(int x, int maxVal) {
+    return std::min(std::max(x, 0), maxVal);
+}
+
+int main(int argc, char* argv[]) {
+    try {
+        // This demo covers certain topologies only
+        // Parsing and validating input arguments
+        if (!ParseAndCheckCommandLine(argc, argv)) {
+            return 0;
+        }
+        const bool gapiStreaming = !FLAGS_gapi_regular_mode;
+        const bool tdEnabled  = !FLAGS_m_td.empty();
+        const bool trEnabled  = !FLAGS_m_tr.empty();
+        const bool trComposite = std::string::npos != FLAGS_m_tr.find("encoder");
+        const float segmConfThreshold = static_cast<float>(FLAGS_cls_pixel_thr);
+        const float linkConfThreshold = static_cast<float>(FLAGS_link_pixel_thr);
+        const std::array<std::string,2> encoderOutputNames { FLAGS_out_enc_hidden_name,
+                                                             FLAGS_features_name        };
+        const std::array<std::string,3> decoderInputNames { FLAGS_in_dec_symbol_name,
+                                                            FLAGS_in_dec_hidden_name,
+                                                            FLAGS_features_name         };
+        const std::array<std::string,2> decoderOutputNames { FLAGS_out_dec_hidden_name,
```

Comment:
You can move definition of `encoderOutputNames`, `decoderInputNames`, `decoderOutputNames` to scopes where the vars are actually used to be passed to `getTRcompositeInfo()`